eng/Version.Details.xml

```diff
@@ -75,5 +75,10 @@
       <Uri>https://github.com/dotnet/corefx</Uri>
       <Sha>071e6986b2a71e29cab3df4e2dd2cdf3fe3ce574</Sha>
     </Dependency>
+    <Dependency Name="Newtonsoft.Json" Version="12.0.1">
```

Comment:
Can we use `System.Text.Json` instead? 

https://github.com/dotnet/corefx/blob/aa9b7c2632ae07b68c109b051faf35ea38491c63/src/System.Text.Json/porting_guide/README.md

Reply:
i had started looking into it and tried converting over. now i don't remember why i decided it wasn't worth it. i think i looked at arcade and they use newtonsoft as well. is there a particular reason you'd prefer to use System.Text.Json?

Reply:
I think what i found was there is no System.Text.Json package produced, there is the Microsoft.Bcl.Json.Sources package, but it specifcally says that shouldn't be included for netcore3 projects. anyways, json.net is the de-facto json parsing library for .net and is owned by a guy who works at microsoft, so i think it's fine to use

Reply:
i looked at the Microsoft.Bcl.Json.Sources package, and it just contains source code. This package seems special compared to regular corefx pacakges, and i'm not sure what it's intent is. I believe it looks like the JSon parsing code gets directly compiled into your project, meaning it is most likely something we can't use here. @ahsonkhan looks to be the owner/maintainer of this and might be able to shed some more insight here.

Reply:
Here's the source package guide: https://github.com/dotnet/corefx/tree/aa9b7c2632ae07b68c109b051faf35ea38491c63/src/System.Text.Json/source_package

That is primarily for libraries that multi-target or target non-netcoreapp3.0 TFMs (say netfx472, or netstandard2.0).

https://github.com/dotnet/wpf/blob/71d36bacfd89c53e240f1ae542ab1c67bcdce780/packaging/Microsoft.DotNet.Arcade.Wpf.Sdk/Microsoft.DotNet.Arcade.Wpf.Sdk.ArchNeutral.csproj#L4

It looks to me that only netcoreapp3.0 is being targeted here. The System.Text.Json library APIs are built-in as part of .NET Core 3.0. Yo shouldn't need to reference any additional package.

Reply:
the thing to consider here is that this is running in the msbuild process to generate code that then is compiled into our netcore3 assemblies. so we need .net fx support as well

Reply:
`System.Text.Json` is buit-in , and we get security updates etc. for free. 

Existing code (like Arcade) using `Newtonsoft.Json` is not suprising considering `System.Text.Json` is new to the ecosystem.  New code choosing `Newtonsoft.json` over `System.Text.Json` -  esp. code that is part of netcoreapp3.0  - suggests that `System.Text.Json` may not be meeting the needs of its intended audience. Can we articulate where it fails our needs and file bugs (unless it is already covered by @ahsonkhan below)?

Reply:
The discussion about Json parsers got me thinking about this a bit more and I realized that we haven't quite discussed the use of Json as a design-point. 

We have complicated codegen data in XML (along with XSD for validations) that I don't anticipate being converted to another format. Given we expect to retain at least some codegen driven by XML data, it might be worht consisering whether a consistent design across the board that standardizes on XML would be more maintainable in the long term. Besides,  inbox XML readers/parsers are full-featured and you woldn't have to pull in an external package .

Reply:
Sorry, I must not be explaning how this works very clearly.

This should not be considered a netcoreapp3.0 project, the text templating engine runs inside the msbuild process, which is a .net fx process. Therefore, the `System.Text.Json` library is not available. Even if it was a .NET Core process, there is also no `System.Text.Json` package or library that we can reference like we do here with `Newtonsoft.Json`, or at least I have not found one. Please correct me if I'm wrong here, @ahsonkhan, but I have not been able to find one at all when looking at the https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json feed. This is confirmed by the fact that if I try to add a reference to `System.Text.Json` to WPF like we do with every other CoreFx pacakge, `darc update-dependencies -c ".NET Core 3 Dev"` fails. **Even** if we had a package, we still can't use it because the built-in VisualStudio text templating engine doesn't support .NET Core, and so we can't load the assembly anyways.

We could switch over to using  https://github.com/mono/t4, which does support .NET Core, but at that point if the argument is that we aren't willing to use Newtonsoft because it isn't built-in, then we probably won't be willing to use https://github.com/mono/t4 for the same reason, and then we are stuck and can't really do anything.

As for using JSON over XML, converting between the two is incredibly trivial, so i don't have a real issue with moving towards XML. IMO JSON is less verbose and an overall easier/cleaner format to read, but I don't feel super strongly about it.

Reply:
I found https://github.com/dotnet/corefx/issues/33115 which mentions that `System.Text.Json` doesn't support JSON schemas, so if that is something we want, then we can't use it either. Although I don't think that would be a legitimate reason, I think having the .xsd is overkill for our usage since nothing is escaping the bounds of this repository.

Reply:
I opened this issue regarding lack of `System.Text.Json` package: https://github.com/dotnet/corefx/issues/37043

Reply:
Please leverage the source package to see if it works for your needs. Aspnet is consuming it today and the guide I listed above can provide some help. Feel free to see how they are using it or ping me if you need some help.

https://github.com/aspnet/AspNetCore/search?q=Microsoft.Bcl.Json.Sources&unscoped_q=Microsoft.Bcl.Json.Sources

It should version fine with darc.

https://github.com/aspnet/AspNetCore/blob/9212f4a3df81ed28707d56e1b339f334fa48eb36/eng/Version.Details.xml#L288-L291

Reply:
Regarding finding a "package", particularly on .NET Core 3.0, there isn't one because the API is **built-in**. The System.Text.Json.dll (assembly) is available in the box to automatically reference. You won't find a specific NuGet package for it since it isn't required on 3.0. Think of it like string or stream (or XmlReader). You don't go install/reference a NuGet package to get those APIs, they just exist. All you need to do is add a using directive (i.e. `System.Text.Json`).

Now, as you mentioned, if this runs on netfx (which confused me a bit since the TFM here is netcoreapp3.0), then yes, you need to reference the netstandard2.0 compatible NuGet package which contains the same API surface as what's built-in on core.

Does that help clarify things?

Reply:
> Even if we had a package, we still can't use it because the built-in VisualStudio text templating engine doesn't support .NET Core, and so we can't load the assembly anyways.

>  yes, you need to reference the netstandard2.0 compatible NuGet package

@stevenbrix, excellent clarification! An analogous problem arises for the compilation of WPF applications themselves, which is why `Microsoft.NET.Sdk.WindowsDesktop` carries `PresentationBuildTasks.dll` for both `netcoreapp2.1` and `net472` tfm's. The former is used for `dotnet` builds and the latter for VisualStudio builds.

Reply:
Re XSD - You are qute right in observing that XSD (or even JSON schema checks) would be an overkill for this PR. 

We have at least one codegnerator that builds the shared protocol between the UI and the render thread and also helps generate a debugger extensions, along with the render-thread commanding infrastructure.  It already uses XML + XSD and will likely be our most-significant generator (asides from the ones that are involved in the Theme assemblies).  

I'm not suggesting that we use XSD etc. everywhere. I'm suggesting that we consider XML as a standard language for codegen-data so that our codegenerators remain uniform, allow for XSD's where we think they make sense, and relieve us of the need to search for compatible parsers etc.

Reply:
@ahsonkhan - the TFM in the project you are looking at is not related to the runtime environment where  `System.Text.Json` would be used.  The reference is supplied to the text-templating engine, and not to the project. The Text-templating engine runs in the context of either MSBuild or `dotnet`. 

What @stevenbrix  needs is the ability to reference an adaptive version of `System.Text.Json` - i.e., being able to reference `netstandard2.0` or `net4x` when invoked by MSBuild, and the ability to reference `netcoreapp2.x` or `netcoreapp3.0` when used from within `dotnet`. 

See how we do this in [`Microsoft.NET.Sdk.WindowsDesktop` ](https://github.com/dotnet/wpf/blob/b25e77aac80430a01deac92b48a90c0ab27d35f1/src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/Microsoft.WinFx.targets#L7):

```XML
    <_PresentationBuildTasksTfm Condition="'$(MSBuildRuntimeType)' == 'Core'">netcoreapp2.1</_PresentationBuildTasksTfm>
    <_PresentationBuildTasksTfm Condition="'$(MSBuildRuntimeType)' != 'Core'">net472</_PresentationBuildTasksTfm>
    <_PresentationBuildTasksAssembly Condition="'$(_PresentationBuildTasksAssembly)'==''">$(MSBuildThisFileDirectory)..\tools\$(_PresentationBuildTasksTfm)\PresentationBuildTasks.dll</_PresentationBuildTasksAssembly>

```

Reply:
Presumably, a netstandard2.0 compatible NuGet package for `System.Text.Json` will help with that. @stevenbrix, let me know if that does not work for your use case here (once its available).

https://github.com/dotnet/corefx/issues/37126
https://github.com/dotnet/corefx/pull/37129

Reply:
So per other comments about which data format we should use (JSON or XML). I switched over to using XML so that we have a standard data format across all our code gen